Distributions.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ @file      Distributions.cpp
+ @brief     functions for sampling monte carlo distributions
+ @details   this file contains functions to be used in a Monte Carlo simulation
+            to sample the various distributions arising in reactor physics.
+ @author    Luke Eure
+ @date      January 8, 2016
+*/
+
+#include "Distributions.h"
+
+/*
+  @brief    returns a uniform random number between 0 and 1
+  @return   a double in the range (0, 1)
+*/
+double urand() {
+    return (double) rand() / (double) RAND_MAX;
+}
+
+/*
+  @brief    function that randomly samples an azimuthal angle                   
+  @details  an azimuthal angle is uniformally sampled in [0, 2 pi]              
+  @return   a randomly sampled angle in [0, 2pi] 
+*/
+double sampleAzimuthalAngle() {
+    return 2 * M_PI * urand();
+}
+
+/*
+ @brief     function that randomly samples a polar angle
+ @details   the cosine of the polar angle is uniformally sampled in [-1, 1]
+            and then transformed to the polar angle with an inverse cosine
+ @return    a randomly sampled polar angle in [0, pi]
+*/
+double samplePolarAngle() {
+    double cos_theta = 2 * urand() - 1.0;
+    return acos(cos_theta);
```

Comment:
No need for the `acos()` the whole application can be done in terms of `mu` (`cos_theta`).  It's better to do it that way because that's how anisotropic scattering is done (as moments of Legendre polynomials based on `mu`).  When you need `sin(theta)` later you can get it by simply using `sin(theta) = sqrt(1-mu^2)` (comes from sin^2+cos^2=1)